File: output/sonarqube_analysis/amazonq_solutions/php/results/q26-word-break.php
<?php
/** https://leetcode.com/problems/word-break */
// Problem: Word Break

//Problem Description:
//Given a string s and a dictionary of strings wordDict, return true if s can be segmented into a space-separated sequence of one or more dictionary words.
//Note that the same word in the dictionary may be reused multiple times in the segmentation.

// Constraints:
// 1 <= s.length <= 300
// 1 <= wordDict.length <= 1000
// 1 <= wordDict[i].length <= 20
// s and wordDict[i] consist of only lowercase English letters.
// All the strings of wordDict are unique.

//Code Structure:

class Solution {

    /**
     * @param String $s
     * @param String[] $wordDict
     * @return Boolean
     */
    function wordBreak($s, $wordDict) {
        $dp = array_fill(0, strlen($s) + 1, false);
        $dp[0] = true;

        for ($i = 1; $i <= strlen($s); $i++) {
            for ($j = 0; $j < $i; $j++) {
                if ($dp[$j] && in_array(substr($s, $j, $i - $j), $wordDict)) {
                    $dp[$i] = true;
                    break;
                }
            }
        }

        return $dp[strlen($s)];
    }
}
    

File: output/sonarqube_analysis/amazonq_solutions/python/results/q26-word-break.py
# https://leetcode.com/problems/word-break/
# Problem: Word Break

#Problem Description:
#Given a string s and a dictionary of strings wordDict, return true if s can be segmented into a space-separated sequence of one or more dictionary words.
#Note that the same word in the dictionary may be reused multiple times in the segmentation.

# Constraints:
# 1 <= s.length <= 300
# 1 <= wordDict.length <= 1000
# 1 <= wordDict[i].length <= 20
# s and wordDict[i] consist of only lowercase English letters.
# All the strings of wordDict are unique.

#Code Structure:
class Solution(object):
    def wordBreak(self, s, wordDict):
        """
        :type s: str
        :type wordDict: List[str]
        :rtype: bool
        """
        n = len(s)
        dp = [False] * (n + 1)
        dp[0] = True

        for i in range(1, n + 1):
            for word in wordDict:
                if i >= len(word) and dp[i - len(word)] and s[i - len(word):i] == word:
                    dp[i] = True
                    break

        return dp[n]                                                                                        
    